esmvaltool/utils/cmorizers/obs/cmorize_obs_ESACCI-OC.py

```diff
@@ -0,0 +1,192 @@
+
+    for cube in cubes:
+        if cube.var_name == rawvar:
+            _fix_var_metadata(cube, var_info)
+            _fix_coords(cube)
+            _fix_data(cube, var)
+            _set_global_atts(cube, attrs)
+            _save_variable(
+                cube,
+                var,
+                out_dir,
+                attrs,
+                local_keys=['coordinates'],
+                unlimited_dimensions=['time'],
+            )
+            _fix_attr_cmip5(out_dir, var)
+
+
+def da_coarsen(da, bin):
+    """ Rebin DataArray with user specified value"""
+    data = np.ma.masked_invalid(da.values)
+    lat = da.coords['lat'].values
+    lon = da.coords['lon'].values
+    # rebin
+    dd = data.shape
+    newda = da[:, :dd[1] // bin, :dd[2] // bin]
+    newda.values = data.reshape([dd[0], dd[1] // bin, bin, dd[2] // bin,
+                                 bin]).mean(-1).mean(-2)
+    newda.coords['lat'].values = lat.reshape([dd[1] // bin, bin]).mean(-1)
+    newda.coords['lon'].values = lon.reshape([dd[2] // bin, bin]).mean(-1)
+
+    return newda
+
+
+def merge_data(in_dir, out_dir, raw_info, bin):
+    """Merge all data into a single (regridded) file"""
+
+    var = raw_info['name']
+    file = raw_info['file']
+    do_bin = True if (bin % 2 == 0) & (bin != 0) else False
+    comment = ''
+    merged_file = os.path.join(out_dir, file + '_merged.nc')
+    #TODO remove 1997* here below before final publication
+    in_files = glob.glob(in_dir + '/' + file + '*1997*.nc')
+    for ff in in_files:
+        DS = xr.open_dataset(ff)
+        da = DS[var].sel(lat=slice(None, None, -1))
+        # remove inconsitent attributes
+        for delkey in [
+                'grid_mapping', 'ancillary_variables', 'parameter_vocab_uri'
+        ]:
+            del da.attrs[delkey]
+        #TODO test with xarray coarsen at v0.12+
+        if (do_bin):
+            da = da_coarsen(da, bin)
+        if ff == in_files[0]:
+            newda = da
+            selkey = ['creator_name','creator_url',\
+                    'license','sensor','processing_level']
+            DSmeta = dict((k, DS.attrs[k]) for k in selkey)
+            if (do_bin):
+                comment = ' '.join([
+                    'Data binned using ',
+                    str(bin), 'by',
+                    str(bin), 'cells average'
+                ])
+                DSmeta['BINNING'] = comment
+            continue
+        newda = xr.concat((newda, da), dim='time')
+
+    # save to file
+    DS = newda.to_dataset()
+    for x, y in DSmeta.items():
+        DS.attrs[x] = y
+    #TODO test encoding with xarray at v0.12+
+    encoding = {
+        'lat': {
+            '_FillValue': False
+        },
+        'lon': {
+            '_FillValue': False
+        },
+        'time': {
+            'calendar': 'gregorian'
+        },
+        var: {
+            '_FillValue': 1.e20
+        }
+    }
+    DS.to_netcdf(merged_file, encoding=encoding, unlimited_dims='time')
+
```

Comment:
is all this `xarray` stuff going to be used in the future as well? Might be worth it if we built an `xarray_utils.py` for that is it's gonna keep recurring

Reply:
So far this is done only here to aggregate the single satellite files into a temporary merged one prior to read them with iris. In addition I added a feature to rebin the data with xarray coarsen function that is now available in versions > 0.12.0.  
I'll give you more detail about this choice in the reply to Bouwe's comment here below...